kyuubi-server/src/main/scala/org/apache/kyuubi/engine/KubernetesApplicationOperation.scala

```diff
@@ -101,8 +102,22 @@ class KubernetesApplicationOperation extends ApplicationOperation with Logging {
           debug(s"Successfully got application info by $tag: $info")
           info
         } else {
-          // client mode
-          jpsOperation.getApplicationInfoByTag(tag)
+          // Kyuubi should wait second if pod is not be created
+          submitTime match {
+            case Some(time) =>
+              val elapsedTime = System.currentTimeMillis() - time
+              if (elapsedTime > submitTimeout) {
+                error(s"Can't find target driver pod with tag: $tag, " +
+                  s"has elapsed time: ${elapsedTime}ms (limit timeout: ${submitTimeout}ms)")
+                ApplicationInfo(id = null, name = null, ApplicationState.NOT_FOUND)
+              } else {
+                warn("Wait for driver pod to be created, " +
+                  s"elapsed time: ${elapsedTime}ms, return UNKNOWN status")
+                ApplicationInfo(id = null, name = null, ApplicationState.UNKNOWN)
```

Comment:
better to use PENDING here, to un-surprise users

Reply:
Given that we are already using PENDING to indicate the status of Spark Driver Pod created on Kubernetes, does it give the user the illusion that the driver pod has been submitted successfully?